x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/snapshotlifecycle/SnapshotLifecycleTask.java

```diff
@@ -78,4 +95,82 @@ public void onFailure(Exception e) {
                .filter(policyMeta -> jobId.equals(SnapshotLifecycleService.getJobId(policyMeta)))
                .findFirst());
     }
+
+    private static class WriteJobStatus extends ClusterStateUpdateTask {
+        private static final ToXContent.Params STACKTRACE_PARAMS =
+            new ToXContent.MapParams(Collections.singletonMap(REST_EXCEPTION_SKIP_STACK_TRACE, "false"));
+
+        private final String policyName;
+        private final String snapshotName;
+        private final long timestamp;
+        private final Optional<Exception> exception;
+
+        private WriteJobStatus(String policyName, String snapshotName, long timestamp, Optional<Exception> exception) {
+            this.policyName = policyName;
+            this.snapshotName = snapshotName;
+            this.exception = exception;
+            this.timestamp = timestamp;
+        }
+
+        static WriteJobStatus success(String policyId, String snapshotName, long timestamp) {
+            return new WriteJobStatus(policyId, snapshotName, timestamp, Optional.empty());
+        }
+
+        static WriteJobStatus failure(String policyId, String snapshotName, long timestamp, Exception exception) {
+            return new WriteJobStatus(policyId, snapshotName, timestamp, Optional.of(exception));
+        }
+
+        private String exceptionToString() throws IOException {
+            if (exception.isPresent()) {
+                try (XContentBuilder causeXContentBuilder = JsonXContent.contentBuilder()) {
+                    causeXContentBuilder.startObject();
+                    ElasticsearchException.generateThrowableXContent(causeXContentBuilder, STACKTRACE_PARAMS, exception.get());
+                    causeXContentBuilder.endObject();
+                    return BytesReference.bytes(causeXContentBuilder).utf8ToString();
+                }
+            }
+            return null;
+        }
+
+        @Override
+        public ClusterState execute(ClusterState currentState) throws Exception {
+            SnapshotLifecycleMetadata snapMeta = currentState.metaData().custom(SnapshotLifecycleMetadata.TYPE);
+
+            assert snapMeta != null : "this should never be called while the snapshot lifecycle cluster metadata is null";
+            if (snapMeta == null) {
+                logger.error("failed to record snapshot [{}] for snapshot policy [{}]: snapshot lifecycle metadata is null",
```

Comment:
We might want to log the snapshot name also so the user can go check it out manually